### PR TITLE
Code review request

### PR DESCRIPTION
**Pull request by poglet**
Hi 

Can you 'Code Review' my python scripts (hex_codes.py, state_names.py, word_counter.py).  There are a few problems here and there but ended up having to rush before submission time.

Kind Regards,
Chris Thomas

### CODE REVIEW

**Comment by david-micallef** [comment]
Hey Chris @poglet ,

state_names.py
- STATENAMES dictionary formatted correctly to PEP8
- Printing table of states is string formatted neatly
- .upper is implemented well

hex_codes.py
- COLOURS dictionary very well organised, which makes it easy to read

word_counter.py
- Good use of .extend with the dictionary to store in the list

Overall your code is well written and there is good use of comments. 

Cheers,

David

File: prac_06/box_layout_demo.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        self.title = "Box Layout Demo"
+        self.root = Builder.load_file('box_layout.kv')
+        return self.root
+
+    def handle_greet(self):
+        print('greet')
+        self.root.ids.output_label.text = "Hello " + self.root.ids.input_name.text
+
+    def clear(self):
+        self.root.ids.output_label.text = 'Enter your name'
```
Comment (david-micallef) [inline prac_06/box_layout_demo.py]: I'm not sure if you need this line

File: prac_06/box_layout_demo.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+
+class BoxLayoutDemo(App):
+    def build(self):
+        self.title = "Box Layout Demo"
+        self.root = Builder.load_file('box_layout.kv')
+        return self.root
+
+    def handle_greet(self):
+        print('greet')
```
Comment (david-micallef) [inline prac_06/box_layout_demo.py]: I don't think you need to have to print 'greet' as you already have the text in the .kv file

File: prac_06/convert_m_to_km.kv [hunk 3]
```diff
@@ -0,0 +1,22 @@
+BoxLayout:
+    orientation: 'vertical'
+    BoxLayout:
+        orientation: 'horizontal'
+        TextInput:
```
Comment (david-micallef) [inline prac_06/convert_m_to_km.kv]: You can make your button size smaller by using - size_hint. It is on slide 23 of the powerpoint for this week.

File: prac_06/convert_m_to_km.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+from kivy.app import App
+from kivy.lang import Builder
+from kivy.core.window import Window
+
+
+class Conversion(App):
+    """ This converts m to km """
+    def build(self):
```
Comment (david-micallef) [inline prac_06/convert_m_to_km.py]: You can make the window size with the dimensions you want by using - Window.size = (600, 200). The numbers 600 and 200 are just estimate numbers. Just add it in the function - def build(self):.

**Comment by david-micallef** [comment]
Hey Chris,

I made a couple of suggestions but I had trouble with this weeks prac. 

Cheers,

David

**Comment by lindsaymarkward** [comment]
Do not re-use the same PR for different pracs. Just start a new one each time.

**Pull request closed without merging**